src/core/Akka.TestKit/EventFilter/Internal/EventFilterApplier.cs

```diff
@@ -479,8 +386,21 @@ protected async Task<T> InterceptAsync<T>(
             var leeway = system.HasExtension<TestKitSettings>()
                 ? TestKitExtension.For(system).TestEventFilterLeeway
                 : _testkit.TestKitSettings.TestEventFilterLeeway;
-
-            var timeoutValue = timeout.HasValue ? _testkit.Dilated(timeout.Value) : leeway;
+            
+            // Calculate timeout - if an explicit timeout is provided, use that (after dilating)
+            // Otherwise, if we're in a WithinAsync block use its remaining time
+            // Otherwise use the default leeway
```

Comment:
I don't see `leeway` variable being used anywhere in this method?

Reply:
Yeah I wondered about that too - basically `leeway` is going to conflict with the `WithinAsync` block. Leeway is really supposed to be the epsilon value, but we have the call chain backwards.